using-ide-vscode.md

```diff
@@ -0,0 +1,234 @@
+* Debugging is based on a default launch configuration. To customize this,
+  click the cog at the top of the **Debug Sidebar** to create a `launch.json`
+  file and customise values.
+
+## Fast edit and refresh development cycle
+
+Flutter offers a best-in-class developer cycle enabling you to see the effect
+of your changes almost instantly with the 'hot reload' feature. See
+[Hot reloading Flutter Apps](/hot-reload/) for details.
+
+## Advanced debugging
+
+### Debugging visual layout issues
+
+During a debug session, several debugging commands will be added to the **Command Palette**, including:
+
+* 'Toggle Baseline Painting': Causes each RenderBox to paint a line at each of its
+  baselines.
+
+* 'Toggle Repaint Rainbow': Show rotating colors on layers when repainting.
+
+* 'Toggle Slow Animations': Slow down animations to enable visual inspection.
+
+* 'Toggle Slow-Mode Banner': Hide the 'slow mode' banner even when running a
+  debug build.
+
+
+### Debugging with Observatory
+
+Observatory is an additional debugging and profiling tool presented with an
+html-based UI. For details see the [Observatory page](https://dart-lang.github.io/observatory/).
+
+To open Observatory:
+
+1. Run your app in debug mode.
+1. Run the **Open Observatory** command from the **Command Palette**.
+
+## Editing tips for Flutter code
+
+### Assists & Quick Fixes
+
+Assists are code changes related to a certain code identifier. A number of these
+are available when the cursor is placed on a Flutter widget identifier, as
+indicated by the yellow lightbulb icon. The assist can be invoked by clicking
+the lightbulb, or by using the keyboard shortcut `Ctrl`+`Enter`, as illustrated
+here:
+
+<img src="/images/vscode/assists.png" style="width:467px;height:234px" alt="Code Assists" />
+
+Quick Fixes are similar, only they are shown with a piece of code has an error and they
+can assist in correcting it.
+
+#### Wrap with new widget assist
+This can be used when you have a widget that you want to wrap in a surrounding widget,
+for example if you want to wrap a widget in a `Row` or `Column`.
+ 
+####  Wrap widget list with new widget assist
+Similar to the assist above, but for wrapping an existing list of widgets rather than an
+individual widget.
+ 
+#### Convert child to children assist
+Changes a child argument to a children argument, and wraps the argument value in a list.
+
+### Snippets
+
+Snippets can be used to speed up entering typical code structures. They
+are invoked by typing their 'prefix', and then selecting from the code
+completion window:
+
+<img src="/images/vscode/snippets.png" style="width:706px;height258px" alt="Snippets" />
+
+The Dart extension includes the following snippets:
+
+* Prefix `stless`: Create a new subclass of `StatelessWidget`.
+* Prefix `stful`: Create a new subclass of `StatefulWidget` and it's associated
+ State subclass.
+* Prefix `stanim`: Create a new subclass of `StatefulWidget`, and it's
```

Comment:
not for this PR, but I really don't know how common this template is; we should probably remove from VS Code and IntelliJ

/cc @stevemessick, for thoughts on whether we should keep it